File: 1/sol2.js
import { promises as fs } from 'fs';

const lines = (await fs.readFile('./input.txt', 'utf-8')).toString().split('\n');
const numbers = lines.map(x => parseInt(x));

const accumulated = numbers.reduce(
  (acc, val) => {
    if (isNaN(val)) {
      acc.push(0);
    } else {
      acc[acc.length - 1] += val;
    }
    return acc;
  },
  [0]
);

console.log(
  accumulated
    .sort()
    .slice(-4, -1)
    .reduce((acc, val) => acc + val, 0)
);

File: 1/sol2.py
from functools import reduce

with open('./input.txt') as file:
    lines = [line.strip() for line in file]

numbers = [None if line == '' else int(line) for line in lines]


def sum_calories(acc, v):
    if v is None:
      acc.append(0)
    else:
      acc[- 1] += v
    return acc


calories_per_person = reduce(sum_calories, numbers, [0])
print(reduce(lambda a, b: a+b, sorted(calories_per_person)[-3:]))
